### PR TITLE
Unit test cleanup

### PR DESCRIPTION
**Pull request by mighele**
Various changes to fix all broken unit tests.

### CODE REVIEW

File: plover/dictionary/test_default_dict.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import json
 import unittest
 
-DICT_PATH='plover/assets/dict.json'
+DICT_PATH='plover/assets/dict_unsorted.json'
```
Comment (balshetzer) [inline plover/dictionary/test_default_dict.py]: This doesn't really fix the test. 

The default dictionary now consists of all the .json files combined. The test would need to load all of them rather than just dict_unsorted.json.

File: plover/assets/dict_unsorted.json [hunk 2]
```diff
@@ -121866,7 +121866,6 @@
 "KWRA/AUL": "y'all",
 "KWRAOU/AUL": "y'all",
 "KWRAB/PWA/TKAB/PWA": "yabba dabba",
-"KWRAT": "yacht",
```
Comment (balshetzer) [inline plover/assets/dict_unsorted.json]: Why remove these entries? Are these changes related to fixing unit tests?

File: plover/dictionary/test_loading_manager.py [hunk 3]
```diff
@@ -25,8 +25,6 @@ def __call__(self, filename):
         loader = MockLoader(files)
         with patch('plover.dictionary.loading_manager.load_dictionary', loader):
             manager = loading_manager.DictionaryLoadingManager()
-            manager.start_loading('a')
-            manager.start_loading('b')
```
Comment (balshetzer) [inline plover/dictionary/test_loading_manager.py]: This line tested that we only loaded file 'b' once even though we tried to load it twice.

File: plover/dictionary/test_loading_manager.py [hunk 4]
```diff
@@ -25,8 +25,6 @@ def __call__(self, filename):
         loader = MockLoader(files)
         with patch('plover.dictionary.loading_manager.load_dictionary', loader):
             manager = loading_manager.DictionaryLoadingManager()
-            manager.start_loading('a')
```
Comment (balshetzer) [inline plover/dictionary/test_loading_manager.py]: This line tests that the start_loading function works on its own. The other dictionaries are loaded with the load function.

File: plover/assets/dict_unsorted.json (hunk 2 above)
Comment (Achim63) [inline plover/assets/dict_unsorted.json]: Please don't delete those - that's exactly how I'd try to write "yacht" so it's a perfectly valid entry. And who doesn't like the Flintstones?

Comment (mighele) [inline plover/assets/dict_unsorted.json]: The reason I've removed KWRAT = yacht is because it conflicts with an older entry for "{^@^}". Since the latter entry was in the dictionary long before the "yacht" entry I decided to remove the yacht one.

File: plover/dictionary/test_loading_manager.py (hunk 3 above)
Comment (mighele) [inline plover/dictionary/test_loading_manager.py]: Ah ok. The start_loading function gets called later on in the normal way so I assumed this was redundant code. I'll re-instate.

File: plover/dictionary/test_default_dict.py (hunk 1 above)
Comment (mighele) [inline plover/dictionary/test_default_dict.py]: True. I just figured it was worth a quick fix so at least the tests don't error.

File: plover/assets/dict_unsorted.json (hunk 2 above)
Comment (Achim63) [inline plover/assets/dict_unsorted.json]: There's no entry for "@" in my release version of the dictionaries ... I had to add it to my personal dictionary. Might be a good idea to add it, though, maybe use "KRWA_Asterisk_T" to leave yacht as is?

Comment (mighele) [inline plover/assets/dict_unsorted.json]: I've just updated this to KWRA\*T.
